core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -89,6 +92,13 @@ private[spark] object JettyUtils extends Logging {
             val result = servletParams.responder(request)
             response.setHeader("Cache-Control", "no-cache, no-store, must-revalidate")
             response.setHeader("X-Frame-Options", xFrameOptionsValue)
+            xXssProtectionValue.foreach(response.setHeader("X-XSS-Protection", _))
+            if (xContentTypeOptionsValue.get.equalsIgnoreCase("true")) {
+              response.setHeader("X-Content-Type-Options", "nosniff")
+            }
+            if (conf.get("spark.ssl.enabled").equalsIgnoreCase("true")) {
```

Comment:
I think you can check `request.scheme` if it is "https" or not?

Reply:
@jerryshao Thanks for the tip. I will do that.